apps/meteor/client/views/composer/EmojiPicker/EmojiPicker.tsx

```diff
@@ -129,11 +129,16 @@ const EmojiPicker = ({ reference, onClose, onPickEmoji }: EmojiPickerProps) => {
 	};
 
 	useEffect(() => {
-		if (recentEmojis.length === 0 && currentCategory === 'recent') {
-			const customEmojiList = getEmojiListsByCategory().filter(({ key }) => key === 'rocket');
-			handleGoToCategory(customEmojiList.length > 0 ? 0 : 1);
+		let unmounted = false;
+
+		if (!unmounted) {
+			setCurrentCategory('recent');
 		}
-	}, [actualTone, recentEmojis, getEmojiListsByCategory, currentCategory, setRecentEmojis]);
+
+		return () => {
+			unmounted = true;
+		};
+	}, [setCurrentCategory]);
```

Comment:
I don't think I understand this hook 🤔 

`unmounted` is declared and assigned `false`. The `if` in `L134` (`if(!unmounted)`) will always enter. 
Then on the cleanup the local variable is set to `true`, but at that point if there's a next render it will re-declare the variable.

Did you mean to set the variable outside the scope of the hook? 
The way it is, wouldn't the following serve the same purpose:
```js
useEffect(() => {
  setCurrentCategory('recent')
}, [setCurrentCategory])
```

Reply:
please, check [this](https://react.dev/learn/synchronizing-with-effects#fetching-data) out